controller/resource_roles.go

```diff
@@ -30,19 +30,16 @@ func (c *ResourceRolesController) ListAssigned(ctx *app.ListAssignedResourceRole
 
 	var roles []role.IdentityRole
 
-	err := application.Transactional(c.db, func(appl application.Application) error {
-		err := appl.ResourceRepository().CheckExists(ctx, ctx.ResourceID)
-		if err != nil {
-			log.Error(ctx, map[string]interface{}{
-				"resource_id": ctx.ResourceID,
-				"err":         err,
-			}, "does not exist")
-			return errors.NewNotFoundError("resource_id", ctx.ResourceID)
-		}
-
-		roles, err = appl.RoleManagementModelService().ListByResource(ctx, ctx.ResourceID)
-		return err
-	})
+	err := c.app.ResourceRepository().CheckExists(ctx, ctx.ResourceID)
```

Comment:
I understand why @sbose78 added this check, because there is no other easy way to return a NotFound error from this controller for an invalid resource otherwise.

Reply:
Agreed, Shane.